sdks/python/apache_beam/ml/inference/api.py

```diff
@@ -15,38 +15,29 @@
 # limitations under the License.
 #
 
+import abc
 from dataclasses import dataclass
-from enum import Enum
 from typing import Tuple
 from typing import TypeVar
 from typing import Union
 
 import apache_beam as beam
 
 
-class PyTorchDevice(Enum):
-  CPU = 1
-  GPU = 2
-
-
-class SklearnSerializationType(Enum):
-  PICKLE = 1
-  JOBLIB = 2
-
-
-@dataclass
 class BaseModelSpec:
-  model_uri: str
-
-
-@dataclass
-class PyTorchModelSpec(BaseModelSpec):
-  device: PyTorchDevice
-
+  """
+  Model factory that returns ModelLoader and
+  InferenceRunner objects to be used
+  """
+  @abc.abstractmethod
+  def get_model_loader(self):
+    "Returns ModelLoader object"
```

Comment:
I think this is good for now.  

Just a note, but I don't think you need to do anything: Going forward it seems to me that there will be an implementation that uses a service (instead of loading model) a that maybe returns none for get_model_loader.  I haven't decided if this should return None by default or if a NotImplementedError is fine.